packages/@aws-cdk/lambda-layer-awscli/lib/awscli-layer.ts

```diff
@@ -1,19 +1,35 @@
-import * as path from 'path';
+import { AwsCliV1Asset } from '@aws-cdk/asset-awscli-v1';
 import * as lambda from '@aws-cdk/aws-lambda';
 import { FileSystem } from '@aws-cdk/core';
+import { AssetSource } from '@aws-cdk/interfaces';
 import { Construct } from 'constructs';
 
+/**
+ * AwsCliLayer construct props
+ */
+export interface AwsCliLayerProps {
+  /**
+   * Use this property to supply your own version of the AWS CLI
+   *
+   * @default - An asset containing AWS CLI v1 will be used.
+   */
+  readonly awsCliVersion?: AssetSource;
```

Comment:
Briefly discussed naming change that might make the type feel more intuitive to users IE: `awsCliSource` since the interface is called `AssetSource`.